components/script/dom/gpudevice.rs

```diff
@@ -1094,3 +1288,46 @@ fn convert_texture_size_to_wgt(size: &GPUExtent3DDict) -> wgt::Extent3d {
         depth: size.depth,
     }
 }
+
+fn get_component_from_format(format: GPUTextureFormat) -> GPUTextureComponentType {
```

Comment:
We [have this](https://github.com/gfx-rs/wgpu/blob/d7d0fd5dfe9382b596ca4f1eff12663691603ca5/wgpu-types/src/lib.rs#L1589), please use it instead :)